src/gui/DatabaseOpenWidget.cpp

```diff
@@ -174,13 +173,8 @@ QSharedPointer<Database> DatabaseOpenWidget::database()
 
 void DatabaseOpenWidget::enterKey(const QString& pw, const QString& keyFile)
 {
-    if (!pw.isEmpty()) {
-        m_ui->editPassword->setText(pw);
-    }
-    if (!keyFile.isEmpty()) {
-        m_ui->comboKeyFile->setEditText(keyFile);
-    }
-
+    m_ui->editPassword->setText(pw);
+    m_ui->comboKeyFile->setEditText(keyFile);
```

Comment:
I don't like that the field is cleared immediately after hitting enter. That restores the old annoying behaviour where you cannot fix typos without retypting the whole thing. What we should do, however, is reset the display mode to bullets as soon as enter is pressed. This is the same behaviour that KeePass2Android has.

Reply:
You wrote this code :wink:

Reply:
I wrote it so that it won't clear out the field after enter anymore. That's what we had before. ;-)

Reply:
Ah, nvm. I misread. I thought this was resetting the field to empty. We should still reset the display mode, though. That's something I noticed was missing.